src/arviz_plots/plots/utils.py

```diff
@@ -238,16 +243,21 @@ def add_reference_lines(
         ref_kwargs.setdefault("color", "gray")
     if "linestyle" not in ref_aes:
         ref_kwargs.setdefault("linestyle", plot_bknd.get_default_aes("linestyle", 2, {})[1])
-
+    ref_dt, artist_dims = {}, {}
     if isinstance(references, dict):
         for key, value in references.items():
             if key in data_vars:
-                ref_dt = Dataset({key: DataArray(np.asarray(value))})
-                plot_collection.map(
-                    plot_func, "ref_lines", data=ref_dt, ignore_aes=ref_ignore, **ref_kwargs
-                )
+                ref_dt[key] = DataArray(np.asarray(value))
+                artist_dims[key] = np.size(value)
```

Comment:
I have now seen it is not very clear in the docs, but `artist_dims` should be a mapping of dimension name to dimension length. Therefore I don't think it should be named the same as the variable (not sure it will work consistently either repeating variable names and dimension names in xarray, I am a bit surprised it works at all in fact). I would hardcode the dimension name to something like `ref_line_dim`